amr-wind/wind_energy/ABLWallFunction.cpp

```diff
@@ -60,81 +62,194 @@ ABLWallFunction::ABLWallFunction(const CFDSim& sim)
+                    amrex::Abort(
+                        "Cannot find surface_temp_flux_timetable file: " +
+                        m_surf_temp_flux_timetable);
+                }
+                amrex::Real data_time, data_value;
+                while (ifh >> data_time) {
+                    m_surf_temp_flux_time.push_back(data_time);
+                    m_surf_temp_flux_value.push_back(data_value);
+                    ifh.ignore(
+                        std::numeric_limits<std::streamsize>::max(), '\n');
+                }
+            }
+        }
+
+        // - Specified near-surface temperature mode:
+    } else if (pp.contains("near_surface_temp_timetable")) {
+        m_surf_temp_flux = false;
+        m_surf_temp = false;
+        m_nearsurf_temp = true;
+
+        m_nearsurf_temp_use_table = true;
+
+        if (pp.contains("near_surface_height")) {
+            pp.get("near_surface_height", m_mo.zNearSurf);
+        } else {
+            amrex::Abort(
+                "ABLWallFunction: near_surface_height must be specified when "
+                "specifying near-surface temperature.");
+        }
+
+        pp.query("near_surface_temp_timetable", m_nearsurf_temp_timetable);
+
         amrex::Print()
-            << "ABLWallFunction: Surface temperature flux mode is selected."
+            << "ABLWallFunction: Near-surface temperature time table "
+               "mode is selected."
             << std::endl;
-    } else if (pp.contains("surface_temp_timetable")) {
-        pp.query("surface_temp_timetable", m_surf_temp_timetable);
-        m_tempflux = false;
-        m_temp_table = true;
-        amrex::Print() << "ABLWallFunction: Surface temperature time table "
-                          "mode is selected."
-                       << std::endl;
-        if (!m_surf_temp_timetable.empty()) {
-            std::ifstream ifh(m_surf_temp_timetable, std::ios::in);
+
+        if (!m_nearsurf_temp_timetable.empty()) {
+            std::ifstream ifh(m_nearsurf_temp_timetable, std::ios::in);
             if (!ifh.good()) {
                 amrex::Abort(
-                    "Cannot find surface_temp_timetable file: " +
-                    m_surf_temp_timetable);
+                    "Cannot find near_surface_temp_timetable file: " +
+                    m_nearsurf_temp_timetable);
             }
-            amrex::Real data_time;
-            amrex::Real data_value;
-            ifh.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
-            while (ifh >> data_time) {
-                ifh >> data_value;
-                m_surf_temp_time.push_back(data_time);
-                m_surf_temp_value.push_back(data_value);
+            amrex::Real data_time, data_value;
+            while (ifh >> data_time >> data_value) {
+                m_nearsurf_temp_time.push_back(data_time);
+                m_nearsurf_temp_value.push_back(data_value);
+                ifh.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
```

Comment:
This code line is here to skip the header line in the data file. This ignore statement needs to be first, prior to the loop.